### PR TITLE
Fix input mapping. Fixes #144

### LINKED ISSUE #144: [BUG] InputMapper fail to map to correct action
**Description**
When two different input sequences of same length and same end KeyEvent are defined then InputMapper fails to map to correct action.

**To Reproduce**
Define two input sequences as shown in this figure.

Screenshot from 2021-08-09 13-37-02

These two input sequences have two keyEvents with same last KeyEvent. First one is double click on Spotlight device and other one is Next Button Press followed by Click on Spotlight device.

Now try to press these two input sequences one by one, you will not be able to get right action in both cases.

For me, I always get 'b' Key Sequence.

**Expected behavior**
The InputMapper should map to correct action.

### PR DESCRIPTION
**Pull request by jahnf**
Fixes input mapping for bug reported by @mayanksuman in #144.

### CODE REVIEW

File: src/deviceinput.cc [hunk 1]
```diff
@@ -230,52 +202,47 @@ DeviceKeyMap::Result DeviceKeyMap::feed(const struct input_event input_events[],
...
       }
-      else if (i+1 < m_keymaps.size()) // if not last keyevent in seq
-      {
-        const auto r = m_keymaps[i+1].equal_range(kes[i+1]);
-        if (r.first == r.second) continue;
-        refobj->next_events.emplace(&(*r.first));
+
+      // if last item in key event set
+      if (i == kes.size() - 1) {
+        current->action = configItem.second.action;
```
Comment (mayanksuman) [inline src/deviceinput.cc:245]: Here, we are not checking If `action` is empty or not. A user can abuse it to silence some input from device by putting a 'blank' Key Sequence action (empty action) for input sequence.

I agree that this behaviour might be desirable in some cases (as we do not have `No Action` action to silence some Input Sequences).

@jahnf , do we need this behaviour?
 or 
should be check for action being non-empty before adding it to `nextMap` (and provide `No Action` action explicitly)?

Comment (jahnf) [reply src/deviceinput.cc:245]: Yes you are right 👍  - depends on what the use case is .. silence some key sequences might be useful. 
But actually I had some code to check against that (or just ignore the configurations without action) but must have removed it before committing - probably was too late yesterday 😴 
I will add that simple check to ignore configuration items with empty action.

Comment (jahnf) [reply src/deviceinput.cc:245]: The check against an empty shared_ptr is a must, but it is only for sanity reasons - all configs have an action. 
Additionally the last commit on the hidraw feature branch allows empty KeySequenceActions - so basically if you want to "mute" a certain key input on your device -> create an input mapping with input but an empty key sequence.

Comment (mayanksuman) [reply src/deviceinput.cc:245]: Yeah. Hence, we do not need to implement any `No Action` action.

**Pull request merged**